cookbook/controller/argument_value_resolver.rst

```diff
@@ -0,0 +1,230 @@
+    different results. E.g. the ``isVariadic()`` will never return true on
+    PHP 5.5 and only on PHP 7.0 and higher it will give you basic types when
+    calling ``getType()``.
+
+Now that you know what to do, you can implement this interface. To get the
+current ``User``, you need the current security token. This token can be
+retrieved from the token storage::
+
+    namespace AppBundle\ArgumentValueResolver;
+
+    use AppBundle\Entity\User;
+    use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+    use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+    class UserValueResolver implements ArgumentValueResolverInterface
+    {
+        private $tokenStorage;
+
+        public function __construct(TokenStorageInterface $tokenStorage)
+        {
+            $this->tokenStorage = $tokenStorage;
+        }
+
+        public function supports(Request $request, ArgumentMetadata $argument)
+        {
+            if (User::class !== $argument->getType()) {
+                return false;
+            }
+
+            $token = $this->tokenStorage->getToken();
+
+            if (!$token instanceof TokenInterface) {
+                return false;
+            }
+
+            return $token->getUser() instanceof User;
+        }
+
+        public function resolve(Request $request, ArgumentMetadata $argument)
+        {
+            yield $this->tokenStorage->getToken()->getUser();
+        }
+    }
+
+In order to get the actual ``User`` object in your argument, the given value
+must fulfill the following requirements:
+
+* An argument must be typehinted as ``User`` in your action method signature;
+* A security token must be present;
+* The value must be an instance of the ``User``.
+
+When all those requirements are met and true is returned, the ``ArgumentResolver``
+calls ``resolve()`` with the same values as it called ``supports()``.
+
+That's it! Now all you have to do is add the configuration for the service
+container. This can be done by tagging the service with ``kernel.argument_resolver``
+and adding a priority.
+
+.. note::
+
+    While adding a priority is optional, it's recommended to add one to
+    make sure the expected value is injected. The ``ArgumentFromAttributeResolver``
+    has a priority of 100. As this  one is responsible for fetching attributes
+    from the ``Request``, it's also recommended to trigger your custom value
+    resolver with a lower priority. This makes sure the argument resolvers
+    are not triggered in (e.g.) subrequests if you pass your user along:
+    ``{{ render(controller('AppBundle:User:index', {'user', app.user})) }}``.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.value_resolver.user:
+                class: AppBundle\ArgumentValueResolver\UserValueResolver
+                arguments:
+                    - '@security.token_storage'
```

Comment:
@theofidry we recently updated most double quotes in Yaml to single quotes (for consistency). 

Using double quotes (or advocating using them) leads to problems with certain strings that are valid in PHP double quotes. For instance, `"AppBundle\Controller\StaticController"` is a valid PHP value, but it would result in a parser error in Yaml as `\C` and `\S` are not valid character sequences.